x/multisig/handlers_test.go

```diff
@@ -0,0 +1,247 @@
+package multisig
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/iov-one/weave/store"
+
+	"github.com/iov-one/weave"
+	"github.com/iov-one/weave/x"
+)
+
+var (
+	newTx   = x.TestHelpers{}.MockTx
+	helpers = x.TestHelpers{}
+)
+
+// newContextWithAuth creates a context with perms as signers and sets the height
+func newContextWithAuth(perms ...weave.Condition) (weave.Context, x.Authenticator) {
+	ctx := context.Background()
+	// Set current block height to 100
+	ctx = weave.WithHeight(ctx, 100)
+	auth := helpers.CtxAuth("authKey")
+	// Create a new context and add addr to the list of signers
+	return auth.SetConditions(ctx, perms...), auth
+}
+
+// newSigs creates an array with addresses from each condition
+func newSigs(perms ...weave.Condition) [][]byte {
+	// initial addresses controlling contract
+	var sigs [][]byte
+	for _, p := range perms {
+		sigs = append(sigs, p.Address())
+	}
+	return sigs
+}
+
+// queryContract queries a contract from the bucket and handles errors
+// so you get a strongly typed object or a test failure
+func queryContract(t *testing.T, db weave.KVStore, bucket ContractBucket, id []byte) Contract {
```

Comment:
this is fine. bucket.Get() seems simpler, but I guess this is an easy way to test query at the same time as handler?